src/golang/cmd/server/handler/v2/node_check_result_content_get.go

```diff
@@ -0,0 +1,201 @@
+	args := interfaceArgs.(*nodeResultGetArgs)
+	emptyResp := &nodeResultGetResponse{}
+
+	dag, err := h.DAGRepo.GetByDAGResult(
+		ctx,
+		args.nodeResultID,
+		h.Database,
+	)
+	if err != nil {
+		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error occurred when retrieving workflow dag.")
+	}
+
+	dbMergedNode, err := h.OperatorRepo.GetMergedNode(ctx, args.nodeID, h.Database)
+	if err != nil {
+		return nil, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error reading check node.")
+	}
+
+	dbArtifact, err := h.ArtifactRepo.Get(ctx, dbMergedNode.ArtifactID, h.Database)
+	if err != nil {
+		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error occurred when retrieving artifact result.")
+	}
+
+	execState := shared.ExecutionState{}
+	dbArtifactResult, err := h.ArtifactResultRepo.Get(
+		ctx,
+		args.nodeResultID,
+		h.Database,
+	)
+	if err != nil {
+		if !errors.Is(err, database.ErrNoRows()) {
+			return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error occurred when retrieving artifact result.")
+		}
+		// ArtifactResult was never created, so we mark the artifact as cancelled.
+		execState.Status = shared.CanceledExecutionStatus
+	} else {
+		execState.Status = dbArtifactResult.Status
+	}
+
+	// `dbArtifactResult` is not guaranteed to be non-nil here.
+	if dbArtifactResult != nil && !dbArtifactResult.ExecState.IsNull {
+		execState.FailureType = dbArtifactResult.ExecState.FailureType
+		execState.Error = dbArtifactResult.ExecState.Error
+		execState.UserLogs = dbArtifactResult.ExecState.UserLogs
+	}
+
+	artifactObject := artifact.NewArtifactFromDBObjects(
+		uuid.UUID{}, /* signature */
+		dbArtifact,
+		dbArtifactResult,
+		h.ArtifactRepo,
+		h.ArtifactResultRepo,
+		&dag.StorageConfig,
+		nil, /* previewCacheManager */
+		h.Database,
+	)
+
+	data, isDownsampled, err := artifactObject.SampleContent(ctx)
+	if err != nil {
+		if errors.Is(err, storage.ErrObjectDoesNotExist()) {
+			return emptyResp, http.StatusOK, nil
+		}
+
+		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Failed to retrieve data for the artifact result.")
+	} else if !errors.Is(err, storage.ErrObjectDoesNotExist()) {
+		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Failed to retrieve data for the artifact result.")
+	}
+
+	return &nodeResultGetResponse{IsDownsampled: isDownsampled, Content: data}, http.StatusOK, nil
+}
+
+// This custom implementation of SendResponse constructs a multipart form response with two fields:
+// 1: "metadata" contains a json serialized blob of artifact result metadata.
+// 2: "data" contains the artifact result data blob generated the serialization method
+// specified in the metadata field.
+func (*NodeCheckResultContentGetHandler) SendResponse(w http.ResponseWriter, interfaceResp interface{}) {
```

Comment:
I think you can remove these as metric / check results are small, and we can just use the json serialization to send over the contents